{{cookiecutter.project_slug}}/.github/workflows/ci.yml

```
@@ -87,3 +87,86 @@ jobs:
       shell: bash
       working-directory: ${{ "{{runner.workspace}}" }}/build
       run: ctest
+
+{% if cookiecutter.codecovio == "Yes" %}
+  coverage-test:
+    name: Coverage Testing
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+
+    - name: Install LCov
+      run: |
+        sudo apt-get install -y lcov
+
+{% if cookiecutter.use_submodules == "No" %}
+    - name: Install Catch2 (Linux + MacOS)
+      if: runner.os != 'Windows'
+      run: |
+        git clone -b v2.13.3 https://github.com/catchorg/Catch2.git
+        cd Catch2
+        mkdir build
+        cd build
+        cmake -DBUILD_TESTING=OFF  ..
+        sudo cmake --build . --target install
+
+    - name: Install Catch2 (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        git clone -b v2.13.3 https://github.com/catchorg/Catch2.git
+        cd Catch2
+        mkdir build
+        cd build
+        cmake -DBUILD_TESTING=OFF  ..
+        cmake --build . --target install
+{% if cookiecutter.python_bindings == "Yes" %}
+    - name: Install Pybind11 (Linux + MacOS)
+      if: runner.os != 'Windows'
+      run: |
+        git clone -b v2.6.1 https://github.com/pybind/pybind11.git
+        cd pybind11
+        mkdir build
+        cd build
+        cmake -DBUILD_TESTING=OFF  ..
+        sudo cmake --build . --target install
+
+    - name: Install Pybind11 (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        git clone -b v2.6.1 https://github.com/pybind/pybind11.git
+        cd pybind11
+        mkdir build
+        cd build
+        cmake -DBUILD_TESTING=OFF  ..
+        cmake --build . --target install
+{%- endif %}
+{%- endif %}
+
+    - name: make build directory
+      run: cmake -E make_directory ${{ "{{runner.workspace}}" }}/build
+
+    - name: configure cmake
+      shell: bash
+      working-directory: ${{ "{{runner.workspace}}" }}/build
+      run: cmake $GITHUB_WORKSPACE -DCMAKE_CXX_FLAGS="--coverage"
+
+    - name: build
+      shell: bash
+      working-directory: ${{ "{{runner.workspace}}" }}/build
+      run: cmake --build .
+
+    - name: run tests
+      shell: bash
+      working-directory: ${{ "{{runner.workspace}}" }}/build
+      run: ctest
+
+    - name: collect coverage report
+      shell: bash
+      working-directory: ${{ "{{runner.workspace}}" }}
+      run: |
+        lcov --directory ./build/src --capture --output-file coverage.info
+        bash <(curl -s https://codecov.io/bash) -f coverage.info || echo "Codecov did not collect coverage reports"
```

```suggestion
        bash <(curl --connect-timeout 10 --retry 5 -s https://codecov.io/bash) -f coverage.info || echo "Codecov did not collect coverage reports"
```
this curl command just downloads a bash script, but the codecov.io server can be flaky, so best to retry a few times in case this fails